server/src/main/java/org/apache/cassandra/sidecar/acl/authentication/ReloadingJwtAuthenticationHandler.java

```diff
@@ -72,14 +80,24 @@ public ReloadingJwtAuthenticationHandler(Vertx vertx,
         this.vertx = vertx;
         this.jwtParameters = jwtParameters;
         this.roleProcessor = roleProcessor;
-
-        periodicTaskExecutor.schedule(new OAuth2AuthHandlerGenerateTask());
+        if (jwtParameters.jwtAuthType().equals(JwtParameters.AuthType.STATELESS))
+        {
+            periodicTaskExecutor.schedule(new PeriodicStatelessJwtRefreshTask());
+        }
+        else if (jwtParameters.jwtAuthType().equals(JwtParameters.AuthType.OAUTH))
+        {
+            periodicTaskExecutor.schedule(new OAuth2AuthHandlerGenerateTask());
+        }
+        else
+        {
+            throw new IllegalStateException("Unsupported JWT Auth Type: " + jwtParameters.jwtAuthType());
+        }
     }
 
     @Override
     public void authenticate(RoutingContext context, Handler<AsyncResult<User>> handler)
     {
-        OAuth2AuthHandlerImpl oAuth2AuthHandler = delegateHandler.get();
+        AuthenticationHandlerInternal oAuth2AuthHandler = delegateHandler.get();
```

Comment:
Nit: Rename to more generic variable name

```suggestion
        AuthenticationHandlerInternal authHandler = delegateHandler.get();
```